08_HashTable/HashTable.h

```diff
@@ -97,25 +158,46 @@ class HashTable {
             pos = quadraticProbing(element.first);
             table[pos].first = element.first;
             table[pos].second = false;
-            ++count_add;
+            ++size_fullness;
         }
         table_2.clear();
     }
     int quadraticProbing(const std::string& str) {
         int hash = hash_polinom(str.c_str());
         int index = 0;
         int pos = (hash+index)&(size_hashtable-1);
-        while((table[pos].first!=str) && (!table[pos].second)){
+        int firstfreepos = -1;
+        std::pair<std::string, bool> targetStr{str, false};
+        std::pair<std::string, bool> emptyUnit{"", true};
+        std::pair<std::string, bool> deleteUnit{"", false};
+
+        while(true){
+            if (table[pos]==targetStr)
+                return pos;
+            if (table[pos]==emptyUnit)
+                return pos;
+            if (table[pos]==deleteUnit){
+                firstfreepos=pos;
+                break;
+            }
+            ++index;
+            pos=(pos+index)&(size_hashtable-1);
+        }
+        while(true){
```

Comment:
Зачем появился второй цикл?

Reply:
Я не захотел плодить функции `quadraticProbing`.
Цель ее: возвращать номер ячейки с переданным ключом или первый номер ячейки, куда можно записать ключ.

Первый цикл       
```suggestion
        while(true){
```

Reply:
Если ты не будешь помечать комментарии, как Resolved мне не придется из разворачивать, чтобы прочитать.
Resolved обычно жмет тот, кто задал вопрос в подтверждение, что он получил ответ)
Так будет удобнее.
Автор же просто пишет ответ, или помечает, что исправил (done, ok и т.п.)